pkg/server/apiextensions.go

```diff
@@ -382,6 +382,31 @@ func shallowCopyCRD(in *apiextensionsv1.CustomResourceDefinition) *apiextensions
 	return &out
 }
 
+// decorateCRDWithBinding copy and mutate crd by
+// 1. adding identity annotation
+// 2. terminating status when apibinding is deleting
+func decorateCRDWithBinding(in *apiextensionsv1.CustomResourceDefinition, identity string, deleteTime *metav1.Time) *apiextensionsv1.CustomResourceDefinition {
+	out := shallowCopyCRD(in)
+
+	out.Annotations[apisv1alpha1.AnnotationAPIIdentityKey] = identity
+
+	if deleteTime.IsZero() {
+		return out
+	}
+
+	out.Status.Conditions = make([]apiextensionsv1.CustomResourceDefinitionCondition, len(in.Status.Conditions), len(in.Status.Conditions)+1)
+
+	out.Status.Conditions = append(out.Status.Conditions, in.Status.Conditions...)
+
+	out.DeletionTimestamp = deleteTime.DeepCopy()
+	apiextensionshelpers.SetCRDCondition(out, apiextensionsv1.CustomResourceDefinitionCondition{
```

Comment:
This will take care of appending a new member to the slice if needed, right?

Reply:
removed the setting of capacity.